File: Software Engineering/Java Fundamentals/Advanced Java/03. Files and Directories/Exercise/src/Problem1OddLines/OddLines.java
package Problem1OddLines;

import java.io.*;

/**
 * Created by User on 8.6.2016 г..
 */
public class OddLines {

    public static void main(String[] args) throws IOException {

        File input = new File("resources\\Problem1OddLines\\03_OddLinesIn.txt");
        File output = new File("resources\\Problem1OddLines\\03_OddLinesOut.txt");

        try (BufferedReader reader = new BufferedReader(new FileReader(input));
             BufferedWriter writer = new BufferedWriter(new FileWriter(output))) {

            reader.readLine(); // The first line is apparently not odd because "Line numbers starts from 0."
            String oddLine;
            while ((oddLine = reader.readLine()) != null) {
                writer.write(oddLine);
                writer.newLine();
                reader.readLine();
            }

        }
    }
}

File: Software Engineering/Fundamental Level/Advanced C#/06. Files and Streams/Homework/Problem 1 Odd Lines/OddLines.cs
// Write a program that reads a text file and prints on the console its odd lines. Line numbers starts from 0. Use StreamReader.

using System;
using System.IO;

namespace Problem_1_Odd_Lines
{
    class OddLines
    {
        static void Main()
        {
            StreamReader reader = new StreamReader("ayy lmao.txt");
            using (reader)
            {
                int lineNumber = 0;
                string line = reader.ReadLine();
                while (line != null)
                {
                    if (lineNumber % 2 != 0)
                    {
                        Console.WriteLine("Line {0}: {1}", lineNumber, line);
                    }
                    lineNumber++;
                    line = reader.ReadLine();
                }
            }
        }
    }
}
